Introduces framework (and icons for vox) for icon based species colors (because fuck the color wheel)

Intigracy opened the pull request:
- imageadd: "Vox are now able to choose from 3 colors of skin (the current green has been darkened, a reddish-brown, and gray) in their preferences. Gray is the obvious best choice."
- tweak: "Vox raiders and adminspawned vox now get a random skin color. Admins are able to varedit vox skin colors by changing s_tone and regenerating icons." 

![2016-04-10_21-21-52](https://cloud.githubusercontent.com/assets/4043940/14416730/4a97a764-ff62-11e5-9a24-15cee346dc13.png)
I changed the default Vox sprite.
![](https://cloud.githubusercontent.com/assets/4043940/14416275/dfeda0da-ff5c-11e5-89f3-429edeba3d2b.png)

Species now have a mob reference to the human they're attached to. I needed to do this to make color changing work right.

The skin tone button on character preferences will now output the actual number of your s_tone for nonhumans.

Facial hair recoloring is now restricted to the same species as hair recoloring is.

Makes use of datum/species/handle_post_spawn to change the mob's color after species (and further, icobase) is applied already.

handle_post_spawn and regenerate_icons call updatespeciescolor(). I wonder what that proc does?

Vox icons are now in their own folder in the human_races folder, since there's more than 1.

Updates 2 unused procs in HELPERS in case anyone ever decides to use them.

proc/random_skin_tone() now has a species argument. It needs to use species.name if species is a datum, otherwise just use species.

Fixes vox not correctly creating vox footprints when running over blood.

Stops this from happening:
![2016-04-08_01-52-47](https://cloud.githubusercontent.com/assets/4043940/14378979/a74bcb30-fd2c-11e5-990f-47dcca753e46.png)

Tested as thoroughly as I can. 19 files change woooooo.

9600bauds commented:
uhh

gbasood commented:
>randomly assigned

no pls

Intigracy commented:
If it matters enough I can make it a preference.

How about the sprites?

SangriaMonkey commented:
What's the point?

NigglyWiggly commented:
Do the hairs change colors?

CptWad commented:
make it not random at all and id be ok with it
also green voxchicken when?

Intigracy commented:
You can see in the picture the hairs are the same color still, but fit anyway.

KingCrimsonIJW commented:
I like it a lot
Security is a pain when vox are antags and you can't tell which is which (Unless you are a faggot who uses non chicken names), but make it a preference.
Though random skin tones sounds like a good thing for some reason

9600bauds commented:
I mean, they're always in their hardsuit anyways

SangriaMonkey commented:
There's literally nothing wrong with names that aren't a string of nonsense.

ihadtoregisterforthis commented:
Make them look less dull, maybe keep the beak and arm colors for all of them the same.
And selectable pls

gbasood commented:
Gotta be selectable skin tone.
Unless you want to make a PR about randomizing human skin tone.

SangriaMonkey commented:
For real though make them selectable if you're going to do this.

Intigracy commented:
Will do.

Intigracy commented:
Updated.

CptWad commented:
:+1:

gbasood commented:
>Species now have a mob reference to the human they're attached to. I needed to do this to make color changing work right.

Does the reference get nulled out at any point ever?
Also @PJB3005 @clusterfack Does it **need** to be nulled out?

PJB3005 commented:
@gbasood yes, on deletion of the mob, for example.

FudgePucker commented:
I can't wait to roleplay as a crow from old animated films.

Mmmmm mmm mastah I love me some of your corn

Intigracy commented:
Oh, I forgot about nulling it.

Thanks guys.

Intigracy commented:
It doesn't look like the species even gets nulled at any point in destroy() either.

```
/mob/living/carbon/human/Destroy()
	if(mind && species && (species.name == "Manifested") && (mind in ticker.mode.cult))//manifested ghosts are removed from the cult once their bodies are destroyed
		ticker.mode.update_cult_icons_removed(mind)
		ticker.mode.cult -= mind

	for(var/obj/Overlays/O in obj_overlays)
		returnToPool(O)
	
	obj_overlays = null

	..()

```

Intigracy commented:
nulled it out in destroy()

gbasood commented:
Does that Destroy() ever get called, though?

Intigracy commented on code/modules/mob/living/carbon/human/death.dm:
```diff
@@ -41,9 +41,14 @@
 
+	var/datum/species/s = species
+	if(s)
+		qdel(s)
```
yes

Brolaire commented:
I like it.

PJB3005 commented on code/modules/mob/living/carbon/human/death.dm:
```diff
@@ -41,9 +41,14 @@
+
 	obj_overlays = null
 
+	var/datum/species/s = species
```
Why are you casting this?

As a result of this cast species doesn't get nulled.

Intigracy replied:
Oh, this is human destroy, I can just null it.

Originally I casted it because species can be either text or the species datum, depending on where along the line of setting its species it is.

Gringonius commented:
Finally more validation to call vox shitbirds.

Intigracy commented:
Dis gud 2 go unless anyone else has issues with the code or otherwise.

ihadtoregisterforthis commented:
My only concern is that the original is left the only one with different beak and limbs color.

Intigracy commented:
![untitled](https://cloud.githubusercontent.com/assets/4043940/14416275/dfeda0da-ff5c-11e5-89f3-429edeba3d2b.png)

ihadtoregisterforthis commented:
I was aiming for the new ones to get modified instead. And I think that modifying the old one will get you angry feedback.

To put it on a better way, the two you are introducing are too similar between each other, even if they aren't bad at all by themselves.

Intigracy commented:
I thought of a few more things to do with this, gimme a bit.

Intigracy commented:
Most of the feedback I've recieved on this have people liking the new sprites.

I think I'm again at the point where I'm done making changes, this has been tested as thoroughly as I can.

New things I added last commit, also have updated OP:

Vox traders are now randomly colored.
Adminspawned Vox (and other species if any get icon based coloring in the future) are now randomly colored. Doesn't apply to human skin tone.
Fixed an incorrect check for Vox footprints from blood, should work now..
Added updatespeciescolor() proc.

I trawled through my changes when updating the OP, so everything should be there even if I missed it here.

Shadowmech88 commented:
>vox traders are randomly colored

Shouldn't they be whatever color the player sets in their preferences, assuming they've got a vox character selected?

CptWad commented:
\>Identify vox raiders ckeys by their looks today! Use this handy guide!

Shadowmech88 commented:
I mixed up PRs and thought this one was the one adding the trader job.
Still, "vox trader" != "vox raider."

Intigracy commented:
fixed trader to read raider

Gringonius commented:
With this will it be possible for plasmamen to be able to choose between the current "zombie" and purple skeleton base icons in the near future?

Intigracy commented:
Aye, it will be easier to implement.

Intigracy commented:
tfw no merge

Exxion commented:
Boo, the old green was better

9600bauds commented:
That would be a helpful comment about six days ago

clusterfack commented:
Opinions are like penises, everyone has one but if you aren't showing it to others its just going to waste.

9600bauds commented:
Shit, I've been doing it wrong all this time

Exxion commented:
Well I couldn't exactly see the new color in-game before it was merged
